tests/partition_test.py

```diff
@@ -0,0 +1,155 @@
+        invars=(a,), outvars=(c,), primitive=lax_partition.partition_put_p,
+        bound_subjaxpr=None, params={'partition_id': 1})
+    partition_put_eqn1 = core.JaxprEqn(
+        invars=(b,), outvars=(d,), primitive=lax_partition.partition_put_p,
+        bound_subjaxpr=None, params={'partition_id': 1})
+    add_eqn = core.JaxprEqn(
+        invars=(c, d), outvars=(e,), primitive=lax.add_p,
+        bound_subjaxpr=None, params={})
+
+    partition_jaxpr = core.Jaxpr(
+        constvars=(),
+        invars=(a, b),
+        outvars=(e,),
+        eqns=(partition_put_eqn0, partition_put_eqn1, add_eqn))
+
+    partition_typed_jaxpr = core.TypedJaxpr(
+        jaxpr=partition_jaxpr,
+        literals=(),
+        in_avals=[abstract_arrays.ShapedArray((), onp.float32)] * 2,
+        out_avals=[abstract_arrays.ShapedArray((), onp.float32)])
+
+    jaxpr = core.Jaxpr(
+        constvars=(),
+        invars=(a, b),
+        outvars=(c,),
+        eqns=(core.JaxprEqn(
+            invars=(a, b),
+            outvars=(c,),
+            primitive=lax_partition.partition_p,
+            bound_subjaxpr=None,
+            params={'jaxpr': partition_typed_jaxpr, 'partition_ids': [0, 1]}),))
+
+    cb = xb.make_computation_builder('xla_computation')
+    xla_args = [
+        cb.ParameterFromNumpy(onp.array((), dtype=onp.float32)),
+        cb.ParameterFromNumpy(onp.array((), dtype=onp.float32))
+    ]
+    outs = xla.jaxpr_subcomp(cb, jaxpr, 'cpu', None, (), '', *xla_args)
+    computation = cb.Build(*outs)
+
+    self.assertIn('sharding={maximal device=1}', computation.GetHloText())
+
+  def test_jaxpr(self):
+    @functools.partial(lax_partition.partition, num_partitions=2)
+    def f(x):
+      y = x * x
+      z = 38. + lax_partition.partition_put(y, 1)
+      return z
+
+    jaxpr = api.make_jaxpr(f)(2.)
+    self.assertEqual(
+        '{ lambda  ; a.\n'
+        '  let b = partition[ jaxpr={ lambda  ; a.\n'
+        '                             let b = mul a a\n'
+        '                                 c = partition_put[ partition_id=1 ] b\n'
+        '                                 d = add c 38.0\n'
+        '                             in [d] }\n'
+        '                     partition_ids=(0, 1) ] a\n'
+        '  in [b] }', str(jaxpr))
+
+  def test_xla_computation(self):
+    @functools.partial(lax_partition.partition, num_partitions=2)
+    def f(x):
+      y = x * x
+      z = 38. + lax_partition.partition_put(y, 1)
+      return z
+
+    computation = api.xla_computation(f)(2.)
+    self.assertIn('sharding={maximal device=0}', computation.GetHloText())
+    self.assertIn('sharding={maximal device=1}', computation.GetHloText())
+
+  def test_simple(self):
+    if xb.device_count() < 2:
+      self.skipTest('requires two devices')
+
+    @functools.partial(lax_partition.partition, num_partitions=2)
+    def f(x):
+      y = x * x
+      z = 38. + lax_partition.partition_put(y, 1)
+      return z
+
+    self.assertEqual(42., f(2.))
```

Comment:
Also inspect which device the result is on? (here and in the following tests)

Reply:
Done